### PR TITLE
feat: add dependency and env tests

### PR DESCRIPTION
**Pull request by keenahn**
See #1 

I think it got closed by accident. See my last comment on #1 to see my latest changes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{
+  "name": "@invisible/basic-tests",
+  "license": "MIT",
+  "version": "1.0.0",
+  "description": "Basic tests for your project: requiring each file and testing environment variables",
```
Comment (gunar) [inline package.json]: If we add more tests I'm sure we'll forget to update this.
Get rid of everything after the colon.

File: test/checkRequireAllFiles.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+const ig = require('ignore')
+const path = require('path')
+
+const { sync } = require('glob-gitignore')
+const {
+  compact,
+  map,
+} = require('lodash/fp')
+
+const checkRequireAllFiles = () => {
```
Comment (gunar) [inline test/checkRequireAllFiles.js]: I would rename this to `checkFilesRequirable`

File: README.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+```json
+// package.json:
+{
+  "prestart": "check-env-from-sample && check-require-all-files",
+  "start": "node index.js"
+  ...
+}
+```
+
+Note: `check-env-from-sample` requires `dotenv` before checking.
```
Comment (gunar) [inline README.md]: Beautifully written

File: index.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+'use strict'
+
+require('dotenv').config()
+
+const checkEnvFromSample = require('./test/checkEnvFromSample')
+const checkRequireAllFiles = require('./test/checkRequireAllFiles')
```
Comment (gunar) [inline index.js]: I find it weird that the present library finds itself under the `test/` directory.
Would put it in `lib/`.

File: scripts/posttest.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
```
Comment (gunar) [inline scripts/posttest.sh]: You might want to use `-e` here

https://stackoverflow.com/questions/9952177/whats-the-meaning-of-the-parameter-e-for-bash-shell-command-line

File: scripts/posttest.sh [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+nsp check
+status=( $status + $? )
+
+if [ $status -eq 0 ]
+then
+  exit 0
+else
+  exit 1
+fi
```
Comment (gunar) [inline scripts/posttest.sh]: You might want to use `-e`, to get rid of this conditional

https://stackoverflow.com/questions/9952177/whats-the-meaning-of-the-parameter-e-for-bash-shell-command-line

File: test/checkEnvFromSample.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+const path = require('path')
+const appRootDir = require('app-root-dir').get()
+const checkEnv = require('check-env')
+const { keys } = require('lodash/fp')
+
+const checkEnvFromSample = () => {
+  const envSamplePath = path.join(appRootDir, 'env.sample')
+  assert(fs.existsSync(envSamplePath), 'env.sample not found')
+  const str = fs.readFileSync(envSamplePath)
+  const config = dotenv.parse(str)
```
Comment (gunar) [inline test/checkEnvFromSample.js]: smurt

File: test/checkRequireAllFiles.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  const getFilenameIfUnrequirable = filename => {
+    const fullPath = path.join(appRootDir, filename)
+    try {
+      require(fullPath)
+      return
+    } catch (err) {
+      return fullPath
+    }
+  }
+  const unrequirable = compact(map(getFilenameIfUnrequirable)(files))
```
Comment (gunar) [inline test/checkRequireAllFiles.js]: very descriptive and easy to reason about

File: scripts/posttest.sh (hunk 6 above)
Comment (keenahn) [reply scripts/posttest.sh]: <iframe src="https://giphy.com/embed/3oz8xLd9DJq2l2VFtu" width="480" height="287" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="https://giphy.com/gifs/election2016-donald-trump-election-2016-3oz8xLd9DJq2l2VFtu">via GIPHY</a></p>

Comment (keenahn) [reply scripts/posttest.sh]: giphy 3

Comment (keenahn) [reply scripts/posttest.sh]: The purpose of the script is exactly NOT to exit early if one of the tests fails. We want to run all of them and output the results at the end.

**Pull request merged**